shell/app-shell/lib/arcs-utils.js

```diff
@@ -88,24 +88,31 @@ const ArcsUtils = {
     return JSON.parse(JSON.stringify(object));
   },
   async createOrUpdateHandle(arc, remoteHandle, idPrefix) {
-    let {metadata, values} = remoteHandle;
+    const {metadata, values} = remoteHandle;
     // construct type object
-    let type = ArcsUtils.typeFromMetaType(metadata.type);
+    const type = ArcsUtils.typeFromMetaType(metadata.type);
     // construct id
-    let id = ArcsUtils.getContextHandleId(type, metadata.tags, idPrefix);
+    const id = ArcsUtils.getContextHandleId(type, metadata.tags, idPrefix);
+    const tag = ArcsUtils.getContextHandleTag(metadata.tags, idPrefix);
     // find or create a handle in the arc context
-    let handle = await ArcsUtils._requireHandle(arc, type, metadata.name, id, metadata.tags);
+    const handle = await ArcsUtils._requireHandle(arc, type, metadata.name, id, [tag]); //metadata.tags);
     await ArcsUtils.setHandleData(handle, values);
     return handle;
   },
   // Returns the context handle id for the given params.
   getContextHandleId(type, tags, prefix) {
     return ''
       + (prefix ? `${prefix}_` : '')
-      + (`${type.toString().replace(' ', '-')}_`).replace(/[\[\]]/g, '!')
+      //+ (`${type.toString().replace(' ', '-')}_`).replace(/[\[\]]/g, '!')
```

Comment:
We don't need to do this any more? Fine to keep around if you'd like, could be worth a comment re: what it used to be needed for and why we think we don't need/want it now. I think in general a small comment with stuff like this would help folks b/c otherwise they never know whether it was accidentally left in nor what it used to be used for. I know you don't leave things accidentally, but others have no context on the history/backstory, so for bus-death risk management, some crumbs considered valuable.